test/integration/EndToEnd.t.sol

```diff
@@ -1,56 +1,58 @@
-import {UpdateRestrictionMessageLib} from "src/hooks/libraries/UpdateRestrictionMessageLib.sol";
-
-import {FullDeployer, FullActionBatcher, CommonInput} from "script/FullDeployer.s.sol";
-
-import {MockValuation} from "test/common/mocks/MockValuation.sol";
-import {MockSnapshotHook} from "test/hooks/mocks/MockSnapshotHook.sol";
-import {LocalAdapter} from "test/integration/adapters/LocalAdapter.sol";
+import {LocalAdapter} from "./adapters/LocalAdapter.sol";
+
+import {ERC20} from "../../src/misc/ERC20.sol";
+import {D18, d18} from "../../src/misc/types/D18.sol";
+import {IAuth} from "../../src/misc/interfaces/IAuth.sol";
+import {CastLib} from "../../src/misc/libraries/CastLib.sol";
+import {MathLib} from "../../src/misc/libraries/MathLib.sol";
+import {ETH_ADDRESS} from "../../src/misc/interfaces/IRecoverable.sol";
+import {IdentityValuation} from "../../src/misc/IdentityValuation.sol";
+
+import {MockValuation} from "../common/mocks/MockValuation.sol";
+
+import {Root} from "../../src/common/Root.sol";
+import {Gateway} from "../../src/common/Gateway.sol";
+import {Guardian} from "../../src/common/Guardian.sol";
+import {PoolId} from "../../src/common/types/PoolId.sol";
+import {GasService} from "../../src/common/GasService.sol";
+import {AccountId} from "../../src/common/types/AccountId.sol";
+import {ISafe} from "../../src/common/interfaces/IGuardian.sol";
+import {IAdapter} from "../../src/common/interfaces/IAdapter.sol";
+import {PricingLib} from "../../src/common/libraries/PricingLib.sol";
+import {ShareClassId} from "../../src/common/types/ShareClassId.sol";
+import {AssetId, newAssetId} from "../../src/common/types/AssetId.sol";
+import {VaultUpdateKind} from "../../src/common/libraries/MessageLib.sol";
+import {MAX_MESSAGE_COST} from "../../src/common/interfaces/IGasService.sol";
+
+import {Hub} from "../../src/hub/Hub.sol";
+import {Holdings} from "../../src/hub/Holdings.sol";
+import {Accounting} from "../../src/hub/Accounting.sol";
+import {HubRegistry} from "../../src/hub/HubRegistry.sol";
+import {ShareClassManager} from "../../src/hub/ShareClassManager.sol";
+
+import {Spoke} from "../../src/spoke/Spoke.sol";
+import {IVault} from "../../src/spoke/interfaces/IVault.sol";
+import {BalanceSheet} from "../../src/spoke/BalanceSheet.sol";
+import {UpdateContractMessageLib} from "../../src/spoke/libraries/UpdateContractMessageLib.sol";
+
+import {SyncManager} from "../../src/vaults/SyncManager.sol";
+import {VaultRouter} from "../../src/vaults/VaultRouter.sol";
+import {IBaseVault} from "../../src/vaults/interfaces/IBaseVault.sol";
+import {IAsyncVault} from "../../src/vaults/interfaces/IAsyncVault.sol";
+import {AsyncRequestManager} from "../../src/vaults/AsyncRequestManager.sol";
+import {IAsyncRedeemVault} from "../../src/vaults/interfaces/IAsyncVault.sol";
+
+import {MockSnapshotHook} from "../hooks/mocks/MockSnapshotHook.sol";
+
+import {FreezeOnly} from "../../src/hooks/FreezeOnly.sol";
```

Comment:
Here the mock appears in the middle of two src imports

Reply:
Here it happens because the first filter is `../` > `../src` for the `hooks` block.